parser/unparser.go

```diff
@@ -230,7 +224,7 @@ func (un *unparser) visitCallUnary(expr *exprpb.Expr) error {
 		return fmt.Errorf("cannot unmangle operator: %s", fun)
 	}
 	un.str.WriteString(unmangled)
-	return un.visit(args[0])
+	return un.visitMaybeParenExpr(args[0])
```

Comment:
Thanks for catching this! When the unary argument is a non-unary operator, the parentheses are incorrect. The current check will add parentheses in around standard function calls as well, so perhaps the best choice would be to check whether `args[0]` is an operator:

```
if arg.GetCallExpr() == nil {
  return false
}
fn := arg.GetCallExpr().GetFunction()
_, nested := operators.FindReverse(fn)
un.visitMaybeNested(arg, nested)
```

Reply:
Prefer using the recommended `isComplexOperator` function in the previous comment with the `maybeVisitNested` call here.